src/pybind/mgr/dashboard/services/ceph_service.py

```diff
@@ -278,6 +279,11 @@ def get_smart_data_by_host(hostname):
                 if device['devid'] not in smart_data:
                     smart_data.update(
                         CephService._get_smart_data_by_device(device))
+            if not smart_data:
+                msg = '[SMART] could not find a device to retrieve SMART data from on host %s'
```

Comment:
It's probably just me being dense here, but is the error message correct? Is it really about not finding a device or not being able to receive data from a given device?

Reply:
Maybe this message is more understandable:
```suggestion
                msg = '[SMART] no device delivered SMART data on host %s'
```

Reply:
Well, the information is correct. At this point, SMART data is and can only be retrieved, when a device is found. And it can happen that Ceph cannot find a device. That depends on which environment is used. I've had that already. We also have issues with inconclusive logs, possibly because the logs added in this PR are missing.

The suggested message would only be correct inside `CephService._get_smart_data_by_device(device))` method. But `_get_smart_data_by_device` has also various detailed log messages for various issues already and does not need to be extended.

In case `_get_smart_data_by_device` is not called at all, not SMART data would be returned but also the appropriate logs would be missing. That's why I think the message is correct and should stay as it is. It points to the problem.

Reply:
After looking at the code again in detail it turned out that logging at that point is not even necessary. The `_get_smart_data_by_device` method is called in any case once, provided a device was returned. If not device is returned, another message will be logged. I therefore removed the logging at that point as I think the `_get_smart_data_by_device` sufficiently logs any issues.